maven-model-builder/src/main/java/org/apache/maven/model/validation/DefaultModelValidator.java

```diff
@@ -829,13 +831,13 @@ private boolean validateId( String fieldName, ModelProblemCollector problems, Se
         }
         else
         {
-            boolean match = ID_REGEX.matcher( id ).matches();
-            if ( !match )
+            if ( !org.apache.commons.lang3.StringUtils.containsOnly( id, ID_CHARS ) )
             {
                 addViolation( problems, severity, version, fieldName, sourceHint,
                               "with value '" + id + "' does not match a valid id pattern.", tracker );
```

Comment:
We changed an internal implementation detail, I don't think that matters to the user. So we shouldn't change the message just because of that. If you think the message is bad in general, we can of course improve it (e.g. say what's allowed in an ID).

Reply:
Agreed.